pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -399,6 +400,28 @@ public synchronized PinotResourceManagerResponse updateInstance(String instanceI
     }
   }
 
+  /**
+   * Updates the tags of the specified instance ID
+   */
+  public synchronized PinotResourceManagerResponse updateInstanceTags(String instanceIdToUpdate, String tags) {
+    InstanceConfig instanceConfig = getHelixInstanceConfig(instanceIdToUpdate);
+    if (instanceConfig == null) {
+      return PinotResourceManagerResponse.failure("Instance " + instanceIdToUpdate + " does not exists");
+    }
+    String[] newTags = tags.split(",");
+    List<String> existingTags = Lists.newArrayList(instanceConfig.getTags());
+    for (String tag : existingTags) {
+      instanceConfig.removeTag(tag);
+    }
+    for (String tag : newTags) {
+      instanceConfig.addTag(tag);
+    }
```

Comment:
Can be simplified as following (not sure why Helix does not provide `setTags()`), also `StringUtils.split()` has better performance than `String.split()` which takes a regex
```suggestion
    List<String> tagList = Arrays.asList(StringUtils.split(tags, ','));
    instanceConfig.getRecord().setListField(InstanceConfig.InstanceConfigProperty.TAG_LIST.name(), tagList);
```

Reply:
Ah thanks for this